apps/challenges/views.py

```diff
@@ -245,6 +246,31 @@ def challenge_detail(request, challenge_host_team_pk, challenge_pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+@api_view(['GET'])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def team_name_for_challenge(request, challenge_pk):
+    """
+    Returns the name of the user's participation team in the current challenge.
+    """
+    if has_user_participated_in_challenge(user=request.user, challenge_id=challenge_pk):
+        participant_team_pk = get_participant_team_id_of_user_for_a_challenge(request.user, challenge_pk)
+        try:
+            participant_team = ParticipantTeam.objects.get(pk=participant_team_pk)
+        except ParticipantTeam.DoesNotExist:
+            response_data = {"error": "ParticipantTeam does not exist"}
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+        serializer = ParticipantTeamDetailSerializer(participant_team)
+        details = serializer.data
+        response_data = {"team_name": details['team_name']}
```

Comment:
@RishabhJain2018 @deshraj should we send participant team as the response from the API and use whatever is need on the frontend? This way if we need to show any other metadata from participant team it will just be a frontend change.

Reply:
Yes, I agree.